x-pack/legacy/plugins/siem/public/hooks/use_ui_settings.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment-timezone';
+
+import { DEFAULT_DATE_FORMAT, DEFAULT_DATE_FORMAT_TZ } from '../../common/constants';
+import { useUiSetting } from '../lib/kibana';
+
+export const useDateFormat = () => useUiSetting<string>(DEFAULT_DATE_FORMAT);
+
+export const useTimeZone = () => {
+  const timeZone = useUiSetting<string>(DEFAULT_DATE_FORMAT_TZ);
+  return timeZone === 'Browser' ? moment.tz.guess() : timeZone;
+};
```

Comment:
Optional but something I try to always do is specify the return type of all functions that I can.

For example this would have string for the return types:

```ts
export const useDateFormat = (): string => useUiSetting<string>(DEFAULT_DATE_FORMAT);

export const useTimeZone = (): string => {
  const timeZone = useUiSetting<string>(DEFAULT_DATE_FORMAT_TZ);
  return timeZone === 'Browser' ? moment.tz.guess() : timeZone;
};
```
 
I know sometimes I don't always remember to do this trick or I cannot for another reason but I do like doing it because it makes it trickier for someone to accidentally introduce a new type.

Optional, no changes, but if you review something of mine and see that I forgot a return type just let me know as I prefer to have them.

Reply:
Interesting! Yeah, with an explicit return type it would be immediately obvious that you changed it, while with an implicit one it would only be caught downstream by consumers (or maybe not at all?). I'll change it!